db/migrations/20200412194822_update_create_back_filled_diff_function.sql

```diff
@@ -0,0 +1,80 @@
+-- +goose Up
+-- SQL in this section is executed when the migration is applied.
+-- +goose StatementBegin
+CREATE OR REPLACE FUNCTION public.create_back_filled_diff(block_height BIGINT, block_hash BYTEA, hashed_address BYTEA,
+                                                          storage_key BYTEA, storage_value BYTEA) RETURNS VOID AS
+$$
+DECLARE
+    last_storage_value  BYTEA := (
+        SELECT storage_diff.storage_value
+        FROM public.storage_diff
+        WHERE storage_diff.block_height <= create_back_filled_diff.block_height
+          AND storage_diff.hashed_address = create_back_filled_diff.hashed_address
+          AND storage_diff.storage_key = create_back_filled_diff.storage_key
+        ORDER BY storage_diff.block_height DESC
+        LIMIT 1
+    );
+    empty_storage_value BYTEA := (
+        SELECT '\x0000000000000000000000000000000000000000000000000000000000000000'::BYTEA
+    );
+BEGIN
+    IF last_storage_value = create_back_filled_diff.storage_value THEN
```

Comment:
Couldn't you add `last_storage_value != create_back_filled_diff.storage_value` to the SELECT query's WHERE clause above?

Reply:
I _think_ if we did that then we'd be at risk of creating duplicates. We need to avoid inserting if the most recent row (previous to the new one) for the given address/key has the same value. If we query `where last_value != new value` then I think we'd skip over more recent rows where the value was the same - but I may not be reasoning through things correctly 🤔

Reply:
Yeah I shink I see - last_storage_value could be null but pass through.